### ISSUE TITLE
support test-unit

### ISSUE THREAD

**Issue by stephanos** [briandunn/flatware#38]
Just throwing it out there :) I would love to use this, but we use [test-unit](https://github.com/test-unit/test-unit). What do you think, how difficult is it to integrate?

**Comment by briandunn** [comment]
Well there are 3 main pieces that have to be built to make a new runner.

## dispatcher

This has to take cli options or the directory structure or whatever and split it into actionable jobs. In the rspec and cucumber dispatchers these jobs correspond to files, but the granularity is arbitrary.

## worker

This has to accept jobs, run those tests, and generate result and progress messages. This probably mostly entails writing a custom formatter that sends messages over sockets rather than printing to standard out. Another part of the worker is making sure that the whole world isn't reloaded between runs. Booting rails between tests will quickly eat all the gains of parallel execution.

## formatter

This collects the messages sent by the worker and prints green dots or errors.

So in short, difficult? Probably not. Depends on the test-unit APIs. The process coordination and communication is all in place, only the adapting is required. Time consuming? Yes indeed.

**Comment by briandunn** [comment]
Parallel testing for `test-unit` is now built into rails. @stephanos are you in a rails context where you can upgrade, or is `test-unit` support still interesting to you?
